components/scream/src/share/field/field_repository.hpp

```diff
@@ -240,20 +244,48 @@ FieldRepository<RealType>::get_field (const std::string name, const std::string
   const auto& map = m_fields.find(name);
   EKAT_REQUIRE_MSG(map!=m_fields.end(), "Error! get_field: " + name + " not found.\n");
   //  Search subset of field repo for matching gridname.
-  for (const auto& it : map->second)
-  {
+  for (const auto& it : map->second) {
     if ( it.first.get_grid_name()==grid) {f_matches.push_back(it.first);}
   }
   // Check to make sure a) the field was found on this grid, and b) only once.
-  EKAT_REQUIRE_MSG(f_matches.size()!=0, "Error! get_field: " + name + " not found on grid " + grid + ".\n");
   EKAT_REQUIRE_MSG(f_matches.size()==1, "Error! get_field: " + name + " found " + std::to_string(f_matches.size()) + " matches on grid " + grid + ".\n");
   // Use this field id to grab field itself.
   return get_field(f_matches[0]);
 }
 
+template<typename RealType>
+std::set<typename FieldRepository<RealType>::field_type>
+FieldRepository<RealType>::
+get_field_group (const std::string& group_name, const std::string& grid_name) const {
```

Comment:
Does this mean that each field group contains fields that are all defined on a single grid?

Reply:
Well, if you declare that field F belongs to group G, then all "copies" of F (on all the different grids) that are registered in the repo will have the group G. However, when requesting groups as in or in/out variables, for each group you need to specify the grid you need it on. If group G is needed on grid "foo" and "bar", then one needs to ask for  two group-grid pairs: (G,foo) and (G,bar).

I can add better comments to the field repo class for this.